lib/erb_lint/reporters/junit_reporter.rb

```diff
@@ -1,111 +1,61 @@
 # frozen_string_literal: true
 
-require "rexml/document"
-require "rexml/formatters/pretty"
-
 module ERBLint
   module Reporters
     class JunitReporter < Reporter
+      ESCAPE_MAP = {
+        '"' => "&quot;",
+        "'" => "&apos;",
+        "<" => "&lt;",
+        ">" => "&gt;",
+        "&" => "&amp;",
+      }.freeze
+
+      PROPERTIES = [
+        ["erb_lint_version", ERBLint::VERSION],
+        ["ruby_engine", RUBY_ENGINE],
+        ["ruby_version", RUBY_VERSION],
+        ["ruby_patchlevel", RUBY_PATCHLEVEL.to_s],
+        ["ruby_platform", RUBY_PLATFORM],
+      ].freeze
+
       def preview; end
 
       def show
-        xml = create_junit_xml
-        formatted_xml_string = StringIO.new
-        REXML::Formatters::Pretty.new.write(xml, formatted_xml_string)
-        puts formatted_xml_string.string
-      end
-
-      private
-
-      CONTEXT = {
-        prologue_quote: :quote,
-        attribute_quote: :quote,
-      }
-
-      def create_junit_xml
-        # create prologue
-        xml = REXML::Document.new(nil, CONTEXT)
-        xml << REXML::XMLDecl.new("1.0", "UTF-8")
-
-        xml.add_element(create_testsuite_element)
+        puts %(<?xml version="1.0" encoding="UTF-8"?>)
+        puts %(<testsuite name="erblint" tests="#{@stats.processed_files.size}" failures="#{@stats.found}">)
 
-        xml
-      end
-
-      def create_testsuite_element
-        tests = stats.processed_files.size
-        failures = stats.found
-        testsuite_element = REXML::Element.new("testsuite", nil, CONTEXT)
-        testsuite_element.add_attribute("name", "erblint")
-        testsuite_element.add_attribute("tests", tests.to_s)
-        testsuite_element.add_attribute("failures", failures.to_s)
-
-        testsuite_element.add_element(create_properties)
+        puts %(  <properties>)
+        PROPERTIES.each do |key, value|
+          puts %(    <property name="#{key}" value="#{value}"/>)
```

Comment:
There's little risk but we should still be escaping all the attributes we pass IMO.